mitodl/mit-open
# Adds learner testimonials support

### What are the relevant tickets?

Closes mitodl/hq#4200

### Description (What does it do?)

Adds the learner testimonials API to the app and the necessary database support for it.

The data model includes:
- Attestant name, title, and avatar image (with avatar image resizing to medium and small sizes)
- The attestation itself
- An optional cover image
- A foreign key for applicable channels
- An optional publish date

The API is exposed at `/api/v0/testimonials/` and includes filtering for channel and/or publish status (published or not).
- Channel allows for an array of channel IDs
- Publish status is a boolean; setting to True delivers results that either have no publish date or have one in the past

### How can this be tested?

Automated tests should pass. In addition, the OpenAPI schema should include the endpoints for v0 of the testimonials api.

Create a handful of testimonials through Django Admin. Each should save successfully.

List the testimonials using the API (which you should be able to access via either the Swagger or Redoc UIs). Each testimonial should have the proper data in it, and there should be a medium and small image for the avatar images you uploaded. Clicking the URLs provided should work.

Filter the API results by specifying either a set of channels or the published flag. The appropriate data should be displayed. Applying both filters should work as well. 

Retrieve a single testimonial by using the retrieve API. This should also work. 

None of the APIs should require authentication.
